app/dot/cpp/core/services/LoginService.java

```diff
@@ -116,22 +115,22 @@ private Jws<Claims> getJwsClaims(String jwtToken) throws LoginException {
       return Jwts.parserBuilder().setSigningKey(key).build().parseClaimsJws(jwtToken);
     } catch (Exception e) {
       logger.info("JWT error {}", e.getMessage());
-      throw new LoginException(Error.INVALID_JWT);
+      throw new LoginException(ErrorCodes.INVALID_JWT.getCode());
     }
   }
 
   public User authorizeRequest(String accessToken, List<UserRole> permittedUserRoles)
-      throws LoginException, UserException, EntityNotFoundException {
+      throws LoginException {
     final String userId = checkJwtAndGetUserId(accessToken);
 
     return userService.userIsActiveAndHasRole(userId, permittedUserRoles);
   }
 
-  public JsonObject refreshTokens(String refreshToken) throws LoginException {
+  public JsonObject refreshTokens(String refreshToken) throws BaseException {
 
     final Session session = sessionRepository.findByField("refreshToken", refreshToken);
     if (session == null) {
-      throw new LoginException(Error.SESSION_NOT_FOUND);
+      throw new BaseException(ErrorCodes.SESSION_NOT_FOUND.getCode());
```

Comment:
Can't we have a wrapper that returns these exceptions since this code is duplicated in so many places? We could have a static method in BaseException that receives an ErrorCodes enum value and returns a new BaseException with that ErrorCode? E.g. throw BaseException.from(ErrorCodes.SESSION_NOT_FOUND)

Reply:
I like that